src/odemis/acq/align/roi_autofocus.py

```diff
@@ -53,34 +53,34 @@ def do_autofocus_in_roi(
     :param ccd: ccd component
     :param focus: focus component
     :param focus_range: focus range, tuple of (zmin, zmax) in meters
-    :param nx: (int) the number of positions in x where to focus
-    :param ny: (int) the number of positions in y where to focus
+    :param focus_points : (x,y) positions in the given bbox where autofocus will be run
     :param conf_level: (0<=float<=1) :param conf_level: (float) cut-off value for confidence level of the focus metric,
         only focus points with a confidence above the cut-off will be saved. Default 0 saves all values.
     :return: (list) list of focus positions in x, y, z
     """
     focus_positions = []
     try:
         init_pos = stage.position.value
-        xmin, ymin, xmax, ymax = bbox
-
-        for y in numpy.linspace(ymin, ymax, ny):
-            for x in numpy.linspace(xmin, xmax, nx):
-                with f._autofocus_roi_lock:
-                    if f._autofocus_roi_state == CANCELLED:
-                        raise CancelledError()
-                    stage.moveAbsSync({"x": x, "y": y})
-                    # run autofocus
-                    f._running_subf = align.AutoFocus(ccd, None, focus, rng_focus=focus_range)
-
-                foc_pos, foc_lev, conf = f._running_subf.result(timeout=900)
-                if conf >= conf_level:
-                    focus_positions.append([stage.position.value["x"],
-                                            stage.position.value["y"],
-                                            focus.position.value["z"]])
-                    logging.debug(f"Added focus with confidence of {conf} at position: {focus_positions[-1]}")
-                else:
-                    logging.debug(f"Focus level is not added due to low confidence of {conf} at position: {x, y}.")
+
+        for points in focus_points:
```

Comment:
"points" is odd with a "s" as it should contain a single point. Actually, I'd suggest to just directly write x & y, like that: `for (x, y) in focus_points:`
You can then drop lines 69 & 70.